integrations/coinbase-integration/src/main/java/org/dash/wallet/integration/coinbase_integration/ui/convert_currency/ConvertViewFragment.kt

```diff
@@ -159,50 +188,86 @@ class ConvertViewFragment : Fragment(R.layout.fragment_convert_currency) {
     }
 
     private fun setAmountValue(pickedCurrencyOption: String, valueToBind: String) {
-        val userAccountData = viewModel.selectedCryptoCurrencyAccount.value
-
-        val cleanedValue =
-            if (viewModel.selectedPickerCurrencyCode !== pickedCurrencyOption &&
-                (viewModel.enteredConvertAmount.toBigDecimalOrNull() ?: BigDecimal.ZERO) > BigDecimal.ZERO
-            ) {
-                when {
-                    (userAccountData?.coinBaseUserAccountData?.balance?.currency == viewModel.selectedPickerCurrencyCode) -> {
-                        if (pickedCurrencyOption == viewModel.selectedLocalCurrencyCode) {
-
-                            valueToBind.toBigDecimal() /
-                                userAccountData.currencyToCryptoCurrencyExchangeRate.toBigDecimal()
-                        } else {
-
-                            valueToBind.toBigDecimal() *
-                                userAccountData.cryptoCurrencyToDashExchangeRate.toBigDecimal()
+        viewModel.selectedCryptoCurrencyAccount.value?.let { userAccountData ->
+
+
+            val cleanedValue =
```

Comment:
This method (`setAmountValue`) seems quite convoluted, and it's all just for getting a clean value. Maybe it's worth to think about how to simplify it / extract some parts to separate methods